### PR TITLE
[PyTorch] Mcore DDP support

### PR DESCRIPTION
**Pull request by ksivaman**

### CODE REVIEW

File: transformer_engine/pytorch/module/layernorm_mlp.py [hunk 1]
```diff
@@ -344,7 +344,7 @@ def forward(
                 dim_size = list(gelu_out.size())
                 dim_size[1] = fc2_weight.size(0)
                 fc2_out = torch.empty(dim_size, dtype=activation_dtype, device=gelu_out.device)
-            _, _, _ = tex.gemm(
+            _ = tex.gemm(
```
Comment (timmoon10) [inline transformer_engine/pytorch/module/layernorm_mlp.py:347]: Nitpick:
```suggestion
            tex.gemm(
```
I see this pattern in a few other places.

**Review by timmoon10 — APPROVED** [review]
LGTM

File: transformer_engine/pytorch/module/layernorm_mlp.py (hunk 1 above)
Comment (ksivaman) [reply transformer_engine/pytorch/module/layernorm_mlp.py:347]: Right, the linter complaints about return value not received for that case

**Pull request merged**